### PR TITLE
Logger + Consumer tag + UnknownHandler

### PR DESCRIPTION
**Pull request by chenrozenes**

### CODE REVIEW

File: lib/client.js [hunk 1]
```diff
@@ -33,6 +33,8 @@ function Client(asyncResponseQueue) {
   this.queueName = undefined;
   // Set Async Response Queue
   this.asyncResponseQueue = asyncResponseQueue;
+  // Optional - specify consumer tag
+  this.consumerTag = consumerTag ? consumerTag + '-client' : undefined;
```
Comment (asafdav) [inline lib/client.js]: why do you add -client?

File: lib/listener.js [hunk 2]
```diff
@@ -3,21 +3,29 @@
...
 /**
  * Start consuming messages
  */
 Listener.prototype.start = function () {
-  console.log('[AMQP] Listener starts');
+  logger.log('[AMQP] Listener starts');
   var that = this;
-  this.channel.consume(this.queue, this.processMsg.bind(this), {noAck: false}).then(function (fields) {
+  var options = { noAck: false };
+  options = !!this.consumerTag ? _.extend(options, { consumerTag: this.consumerTag + '-' + chance.hash({length: 5}) }) : options;
```
Comment (asafdav) [inline lib/listener.js]: I think you shouldn't add the chance.hash logic here, we can let the user decides on his pattern, what if someone wants it to be my-consumer-1 ?

File: lib/client.js (hunk 1 above)
Comment (chenrozenes) [inline lib/client.js]: As we talked, if process has one consumer tag for all its clients and workers I want to separate the clients from workers

File: lib/listener.js (hunk 2 above)
Comment (chenrozenes) [inline lib/listener.js]: I let the user decide, I just concat the hash to ensure uniqueness (if several listeners are created with the same consumer tag in the same process)

**Pull request merged**